### PR TITLE
MU WPCOM: Disable the Open Graph Tags according to the privacy of the site

### LINKED ISSUE #92385: Remove OpenGraph metadata tags from "Coming Soon" sites
### What

If a wpcom site is marked as "Coming Soon", it should output no OpenGraph tags that social media sites can use to generate thumbnails, descriptions, and other previews of the site.

Judging by https://github.com/Automattic/wp-calypso/issues/56779#issuecomment-1230048770, I presume the current behavior is expected, and that a site should instead be set to "Private" to remove this metadata. Hence I'm submitting this as an enhancement and not a bug.

### Why

The front page of a site that is marked as "Coming Soon" still has all the metadata loaded necessary for Slack and Twitter to generate previews of the full site. Here are the OG tags of one such site marked "Coming Soon" with Yoast (`wordpress-seo`) installed and active:

```
    <meta property="og:locale" content="en_US" class="yoast-seo-meta-tag"/>
    <meta property="og:type" content="website" class="yoast-seo-meta-tag"/>
    <meta property="og:title" content="Home - by Nijaeri" class="yoast-seo-meta-tag"/>
    <meta property="og:description" content="Who am I? “Zim, Zima, who got de keys to my bimma?” Well, lol, I’m not Beenie Man. My name is Nijaeri (Nigh-jair-ree). It’s an African American variation of the African name Njeri, and so far, it is 100% unique to me!" class="yoast-seo-meta-tag"/>
    <meta property="og:url" content="https://bynijaeri.com/" class="yoast-seo-meta-tag"/>
    <meta property="og:site_name" content="by Nijaeri" class="yoast-seo-meta-tag"/>
    <meta property="article:modified_time" content="2024-06-04T00:16:07+00:00" class="yoast-seo-meta-tag"/>
    <meta property="og:image" content="https://bynijaeri.com/wp-content/uploads/2024/05/about-nj-1.png" class="yoast-seo-meta-tag"/>
    <meta name="twitter:card" content="summary_large_image" class="yoast-seo-meta-tag"/>
```

Without Yoast:

```
    		<meta name="description" content="Who am I? “Zim, Zima, who got de keys to my bimma?” Well, lol, I’m not Beenie Man. My name is Nijaeri (Nigh-jair-ree). It’s an African American variation of the African name Njeri, and so far, it is 100% unique to me! Read About https://open.spotify.com/artist/4iHNK0tOyZPYnBU7nGAgpQ?si=A3hldOKJQVGe65vMWJFfsA https://open.spotify.com/track/4EI8VuxUuIHKfafU72emqz?si=71b0e3ae964b4268 R&amp;B Memoirs R&amp;B Memoirs View all Posts Be the&hellip;" />

<!-- Jetpack Open Graph Tags -->
<meta property="og:type" content="website" />
<meta property="og:title" content="by Nijaeri" />
<meta property="og:description" content="Music and life those are my genres" />
<meta property="og:url" content="https://bynijaeri.com/" />
<meta property="og:site_name" content="by Nijaeri" />
<meta property="og:image" content="https://bynijaeri.com/wp-content/uploads/2024/05/about-nj-1.png" />
<meta property="og:image:width" content="378" />
<meta property="og:image:height" content="567" />
<meta property="og:image:alt" content="" />
<meta property="og:locale" content="en_US" />
<meta name="twitter:text:title" content="Home" />
<meta name="twitter:image" content="https://bynijaeri.com/wp-content/uploads/2024/05/about-nj-1.png?w=640" />
<meta name="twitter:card" content="summary_large_image" />
```

### How

The site's settings:

by-nijaeri-settings

by-nijaeri-wpcom

Excerpts of how it appears on Slack and Twitter:

by-nijaeri-slack

by-nijaeri-twitter

### PR DESCRIPTION
**Pull request by arthur791004**
Fixes https://github.com/Automattic/wp-calypso/issues/92385

## Proposed changes:

* Add filter to disable the action to print OG tags by either Jetpack or YoastSEO

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Apply changes as below
  * On simple sites, apply changes to wpcom and sandbox your site
  * On atomic sites, apply changes to your site directly
* Go to https://wordpress.com/settings/general/:site
* Set the Privacy to one of the below
  * Coming Soon
  * Public & Prevent third-party sharing
* Open your site with the Incognito mode
* Make sure the OG tags are removed
* If your site is an atomic site, then install YoastSEO and make sure the OG tags of YoastSEO are removed either

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php
Comment (taipeicoder) [inline projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php]:
```suggestion
	if ( ! (bool) get_option( 'wpcom_public_coming_soon' ) && ! (bool) get_option( 'wpcom_data_sharing_opt_out' ) ) {
```

**Comment by taipeicoder** [comment]
Coming soon:
* Without Yoast SEO - I still see the OG tags.
* With Yoast SEO - No OG tags.

Public & Prevent third-party sharing:
* Without Yoast SEO - I still see the OG tags.
* With Yoast SEO - No OG tags.

**Comment by taipeicoder** [comment]
`jetpack_og_tags` seems to be undefined, maybe it hasn't been declared yet?  😕

**Comment by fushar** [comment]
Maybe we should merge this first? Not sure if it's a dependency

- https://github.com/Automattic/jetpack/pull/39010

**Comment by arthur791004** [comment]
> jetpack_og_tags seems to be undefined, maybe it hasn't been declared yet? 😕

I found you have to enable either `publicize` or `sharedaddy` module to see the feature

**Comment by arthur791004** [comment]
> Without Yoast SEO - I still see the OG tags.

It should be resolved by [4fe84c6](https://github.com/Automattic/jetpack/pull/39012/commits/4fe84c6c3f9e94bf43e5cc1328a580818f8920f6)

**Review by taipeicoder — DISMISSED** [review]
Coming soon:

- ✅ Without Yoast SEO - No OG tags.
- ✅ With Yoast SEO - No OG tags.

Public & Prevent third-party sharing:

- ✅ Without Yoast SEO - No OG tags.
- ✅ With Yoast SEO - No OG tags.

File: projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php
Comment (mdawaffe) [inline projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php]: I don't think `wpcom_data_sharing_opt_out` is the right thing to check here. We don't, for example, turn off search engine access or social media expanded links if that setting is true.

**Review by mdawaffe — COMMENTED** [review]
I wonder if `wpcom_public_coming_soon` has other implications that Jetpack should be aware of or take into account itself. I don't think that's a blocker for this PR.

I think you should get a review from someone on a Jetpack team for this, though. It's been a while since I've been well-informed enough to have a trustworthy opinion :)

File: projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php
Comment (arthur791004) [reply projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php]: The value of `wpcom_data_sharing_opt_out ` is true when the `Prevent third-party sharing for your site` is checked. As a result, I think it would be better to avoid sharing on social media as well.

**Comment by arthur791004** [comment]
> I wonder if wpcom_public_coming_soon has other implications that Jetpack should be aware of or take into account itself. I don't think that's a blocker for this PR.

I think we can simply focus on the OG tags in this PR to limit the scope 🙂

File: projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php
Comment (mdawaffe) [reply projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php]: The third parties relevant for this option are Automattic's partners, not social media sites users share their content to. As described in the UI:

> This option will prevent this [public] site’s content from being shared with our licensed network of content and research partners, including those that train AI models.

So turning off OG tags for public sites is outside of the original intent of this setting. I don't think the intent of this setting should expand to include social sharing. If you disagree, could you please make the case in a P2 discussion to get more perspectives involved?

Comment (arthur791004) [reply projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php]: Will limit the changes to the Coming Soon 🙂

Comment (mdawaffe) [reply projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php]: Cool - thanks! Looks like the comment in the lines above needs updating too.

Comment (arthur791004) [reply projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php]: Oh thanks! Updated 👍

File: projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php
Comment (jeherve) [inline projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php]: Would it be an option to simplify this logic and just force the Open Graph off with the `jetpack_enable_open_graph` filter?

I haven't tested but I would expect something like this to work (maybe with some priority changes?)

```suggestion
	add_filter( 'jetpack_enable_open_graph', '__return_false' );
```

If that works, I wouldn't be surprised if Yoast SEO offered a similar filter allowing one to shortcircuit their implementation.

Comment (arthur791004) [reply projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php]: The `check_open_graph` function would [register the hook to turn on the open graph if certain modules are active](https://github.com/Automattic/jetpack/blob/4c74a9d0047e36d85e69278bbd0518651ead2e5b/projects/plugins/jetpack/class.jetpack.php#L1972) so I'm not sure whether it's okay to do that.

However, after re-reading the code, the function also adds the following filter to turn off the open graph if there is any conflict plugin. So, I think I can simply add the same filter.
```php
add_filter( 'jetpack_enable_open_graph', '__return_false', 99 );
```

> If that works, I wouldn't be surprised if Yoast SEO offered a similar filter allowing one to shortcircuit their implementation.

The current way is recommended by https://developer.yoast.com/customization/yoast-seo/disabling-yoast-seo/. We can simply remove all actions for `wpseo_head` but I think it would be better to follow the official suggestion. Does it make sense?

Comment (jeherve) [reply projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php]: > but I think it would be better to follow the official suggestion

Yeah, definitely!

**Review by taipeicoder — APPROVED** [review]
Re-tested and LGTM 👍

**Pull request merged**
